vmdb/app/models/ems_amazon.rb

```diff
@@ -49,13 +49,18 @@ def verify_credentials(auth_type=nil, options={})
     begin
       # EC2 does Lazy Connections, so call a cheap function
       with_provider_connection(options.merge(:auth_type => auth_type)) { |ec2| ec2.regions.map(&:name) }
-    rescue AWS::EC2::Errors::SignatureDoesNotMatch
+    rescue AWS::EC2::Errors::SignatureDoesNotMatch => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "SignatureMismatch - check your AWS Secret Access Key and signing method"
-    rescue AWS::EC2::Errors::AuthFailure
+    rescue AWS::EC2::Errors::AuthFailure => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqHostError, "Login failed due to a bad username or password."
+    rescue AWS::Errors::MissingCredentialsError => err
+      $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
+      raise MiqException::MiqHostError, "Missing credentials"
     rescue Exception => err
       $log.error("MIQ(#{self.class.name}.verify_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
-      raise MiqException::MiqHostError, "Unexpected response returned from system, see log for details"
+      raise MiqException::MiqHostError, "Unexpected response returned from system: #{err.message}"
     end
```

Comment:
I *like* that approach :). Though, I'm not sure about matching on the *text* of the exception in the tests...

But, if we're testing this, you were planning some exception normalization... Do you agree with the kinds of exceptions being thrown now, or do you want to change some? It feels silly to write a test that you'd break by normalizing and you'd have to wonder if those are actually needed like so or don't matter.